asteroid_filterbanks/enc_dec.py

```diff
@@ -283,7 +320,10 @@ def forward(self, spec, length: Optional[int] = None) -> torch.Tensor:
             padding=self.padding,
             output_padding=self.output_padding,
         )
+
         wav = self.filterbank.post_synthesis(wav)
+        start = self.filterbank.kernel_size // 2 if self.center else 0
+        wav = wav[..., start:]
```

Comment:
If we remove the `start` before, the w**2 OLA computation is wrong. So, that's not an option. 
But if we do it after, then we had to divide by w**2 OLA on the sides as well, where it's most likely zero (hence the warning instead of the error). 
Not super satisfied about this, because we'll raise warnings about things that might not be problematic (if we trim the part with zeros after the division, which is the most common).

Reply:
if it is trimmed we can just suppress the warnings. But if reflect padding is used it will be fine ? Isn't reflect padding as you said used in torch stft ?

Reply:
With this design, you cannot know if it's going to be trimmed, that's the problem.. Because you don't have access to the `length` argument. 
`torch.stft` uses reflect padding, yes. But I don't see why this would solve the problem, sorry

Reply:
Forget about the padding. I am not sure I am following. The start seems to be trimmed here right ? So this argument applies only to the tail.

Reply:
Yes, true. 
There is not uncertainty on the beginning of the signal: 0 if not center, else kernel_size // 2
But there is uncertainty on the tail, and `torch.istft`'s approach is weird IMO

Reply:
Is indeed weird in fact. I don't like it, right now you avoid division by w**2 if too small but that could cause an abrupt discontinuity at the tail no ?

Reply:
Yes, completely. 
In torch, they raise an error instead. 
So the with the classic `hanning` window, without center=True, it cannot be `istft`ed :sweat: 

I thought about several cases etc.. And then I realized, if people want to use a "suboptimal" normalization scheme for there stft, they can haha. I won't spend more time than this

Reply:
When we raise a warning, either: 
- Torch would have made an error. 
- Torch wouldn't have made an error, and we'll trim it in the next function. 

Anyway, this is only useful for people who already trained a model with torch.stft and want to transfer the model to ONNX or something. So we can assume that the params are good. Otherwise, I think they should use asteroid's stft instead. Much more practical for transfers, for real-time implementation as well.

Reply:
Agree, also we should write on the docs about this problem so people are aware and can then make their choice.